phpBB/phpbb/notification/manager.php

```diff
@@ -333,10 +333,34 @@ public function add_notifications_for_users($notification_type_name, $data, $not
 		foreach ($this->get_subscription_methods_instances() as $method)
 		{
 			$notified_users = $method->get_notified_users($notification_type_id, array('item_id' => $item_id));
+			$notification_method_name = $method->get_type();
+
 			foreach ($notified_users as $user => $notifications)
 			{
 				unset($notify_users[$user]);
 			}
+
+			/**
+			* Allow filtering the notify_users array by notification_type_name & notification_method_name for a notification that is about to be sent.
+			* Here, $notify_users is already filtered from users who've already been notified.
+			*
+			* @event core.notification_manager_add_notifications_for_users_modify_data
+			* @var	string	notification_type_name		The notification type identifier
+			* @var	string	notification_method_name	The notification method identifier (read only)
+			* @var	array 	data						Data specific for the notification_type_name used will be inserted
```

Comment:
The docblock description above made more sense: `Data specific for this type that will be inserted`